ENH: Add NLO support for aarch64 Linux and macOS

matthewfeickert opened the pull request:
* Resolves #3
* Resolves #10 

## Description

* Add ninja support for linux-aarch64 and osx.
* Bump build number.


<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

matthewfeickert commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13476241365](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13476241365). Examine the logs at this URL for more detail.</sub>

matthewfeickert commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13486286444](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13486286444). Examine the logs at this URL for more detail.</sub>

matthewfeickert commented:
NLO on `osx-64` and `osx-arm64` is still broken.

matthewfeickert committed:
ENH: Add ninja support for linux-aarch64 and osx

* Enable ninja support.
* Remove lock on ninja-hep-ph-static.
   - Removal of --with-looptools allows for ninja-hep-ph v1.2.0
    c.f. https://github.com/peraro/ninja/issues/ 5
* Bump build number.
